### PR TITLE
Improve code coverage

### PR DESCRIPTION
**Pull request by lahirusamith**
## Purpose
$title

Fixed for https://github.com/ballerina-platform/ballerina-standard-library/issues/1656

## Checklist
- [ ] Linked to an issue
- [ ] Updated the changelog
- [ ] Added tests

### CODE REVIEW

**Comment by lahirusamith** [comment]
I will add Array conversion and SSL configuration codes back to again with separate PRs

File: ballerina/types.bal [hunk 1]
```diff
@@ -114,20 +114,6 @@ public distinct class VarrayValue {
     }
 }
 
-
-# Represents XML Oracle DB field.
-#
-# + value - Value of parameter passed into the SQL statement
-public distinct class XmlValue {
```
Comment (niveathika) [inline ballerina/types.bal:121]: Any reason to remove this type?

File: native/src/main/java/io/ballerina/stdlib/oracledb/parameterprocessor/OracleDBStatementParameterProcessor.java [hunk 2]
```diff
@@ -68,19 +67,13 @@ protected void setCustomSqlTypedParam(Connection connection, PreparedStatement p
             case Constants.Types.CustomTypes.VARRAY:
                 setVarray(connection, preparedStatement, index, value);
                 break;
-            case Constants.Types.CustomTypes.NESTED_TABLE:
```
Comment (niveathika) [inline native/src/main/java/io/ballerina/stdlib/oracledb/parameterprocessor/OracleDBStatementParameterProcessor.java:71]: Same

File: native/src/main/java/io/ballerina/stdlib/oracledb/utils/Utils.java [hunk 3]
```diff
@@ -106,51 +102,6 @@ public static Properties generatePoolProperties(BMap clientOptions) {
         return null;
     }
 
-    private static void setSSLConProperties(BMap secureSocket, Properties connProperties) {
```
Comment (niveathika) [inline native/src/main/java/io/ballerina/stdlib/oracledb/utils/Utils.java:109]: We cannot remove functionality just to increase the code coverage. Let's leave the functionality as it is and send another PR is needed to cover this part.

File: ballerina/types.bal (hunk 1 above)
Comment (lahirusamith) [reply ballerina/types.bal:121]: This PR is removed in https://github.com/ballerina-platform/module-ballerinax-oracledb/pull/7/commits/cfe79afd0ea4a9fc4d3242c92c526169d2cfa15d

as it was in a seperate PR from gayal but we could not merge that PR because there a lot of conflicts. 
https://github.com/ballerina-platform/module-ballerinax-oracledb/pull/3

File: native/src/main/java/io/ballerina/stdlib/oracledb/utils/Utils.java (hunk 3 above)
Comment (lahirusamith) [reply native/src/main/java/io/ballerina/stdlib/oracledb/utils/Utils.java:109]: As @Anupama akka mentioned, first task is to increase the 80% of the coverage of the oracledb module and then keep on adding functionality with test cases. But to add ssl test cases will take more time as there must be another docker container to test ssl connections and we have not finalized the docker containers count to be used when running test cases. 

I have create a another issue https://github.com/ballerina-platform/ballerina-standard-library/issues/1672 to track the SSL code and will add the functionality in near future with proper test cases.

Comment (niveathika) [reply native/src/main/java/io/ballerina/stdlib/oracledb/utils/Utils.java:109]: The intention is to increase the code coverage with the current functionalities as soon as possible before working on adding new functionalities. Here the SSL connection functionality is already available. It is not a new feature to remove.

File: ballerina/tests/11-custom-object-type-test.bal [hunk 4]
```diff
@@ -169,126 +138,299 @@ isolated function insertObjectTypeWithEmptyArray() returns sql:Error? {
...
+
+@test:Config {
+    groups: ["query", "object"],
+    enable: false,
+    dependsOn: [selectObjectType]
+}
+isolated function selectObjectTypeNull() returns error? {
+    Client oracledbClient = check new (HOST, USER, PASSWORD, DATABASE, PORT);
+    stream<record{}, error> streamResult = oracledbClient->query(
+        "SELECT pk, col_object FROM TestObjectTypeTable WHERE pk = 15", ObjectRecordType);
```
Comment (daneshk) [inline ballerina/tests/11-custom-object-type-test.bal:282]: Let's always try to go with the type infering from the assigned variable type. As it is our recommendation.

File: ballerina/tests/12-custom-type-varray-test.bal [hunk 5]
```diff
@@ -148,19 +138,268 @@ isolated function insertVarrayWithEmptyArray() returns sql:Error? {
...
+    if (returnData is sql:ApplicationError) {
+        test:assertTrue(returnData.message().includes("Cannot cast varray to type: string[]"),
+            "Incorrect error message");
+    } else {
+        test:assertFail("Querying varray with invalid array type should fail with " +
+                            "sql:ApplicationError");
+    }
+    check streamData.close();
+    check oracledbClient.close();
+}
```
Comment (daneshk) [inline ballerina/tests/12-custom-type-varray-test.bal:405]:
```suggestion
}

```

File: native/src/main/java/io/ballerina/stdlib/oracledb/utils/Utils.java (hunk 3 above)
Comment (daneshk) [reply native/src/main/java/io/ballerina/stdlib/oracledb/utils/Utils.java:109]: Agreed. we can't remove the existing functionalities. It's okay to add test cases in a separate PR, but let's not remove the existing ones.

File: ballerina/types.bal (hunk 1 above)
Comment (daneshk) [reply ballerina/types.bal:121]: @lahirusamith but we want to find out why gayal removed this code. As per his comment in his PR `Oracle JDBC does not provide functionality for XML.`. shall we create an issue, look into it further.

File: ballerina/tests/12-custom-type-varray-test.bal (hunk 5 above)
Comment (lahirusamith) [reply ballerina/tests/12-custom-type-varray-test.bal]: fixed in https://github.com/ballerina-platform/module-ballerinax-oracledb/pull/7/commits/903f0cd0b25dea933b9106c0947ee5a969e13afd

File: ballerina/tests/11-custom-object-type-test.bal (hunk 4 above)
Comment (lahirusamith) [reply ballerina/tests/11-custom-object-type-test.bal]: corrected in https://github.com/ballerina-platform/module-ballerinax-oracledb/pull/7/commits/903f0cd0b25dea933b9106c0947ee5a969e13afd

File: native/src/main/java/io/ballerina/stdlib/oracledb/utils/Utils.java (hunk 3 above)
Comment (lahirusamith) [reply native/src/main/java/io/ballerina/stdlib/oracledb/utils/Utils.java]: added back in https://github.com/ballerina-platform/module-ballerinax-oracledb/pull/7/commits/903f0cd0b25dea933b9106c0947ee5a969e13afd

File: ballerina/types.bal (hunk 1 above)
Comment (lahirusamith) [reply ballerina/types.bal:121]: issue is created https://github.com/ballerina-platform/ballerina-standard-library/issues/1695

File: native/src/main/java/io/ballerina/stdlib/oracledb/parameterprocessor/OracleDBStatementParameterProcessor.java (hunk 2 above)
Comment (lahirusamith) [reply native/src/main/java/io/ballerina/stdlib/oracledb/parameterprocessor/OracleDBStatementParameterProcessor.java:71]: issue was created to track this removal https://github.com/ballerina-platform/ballerina-standard-library/issues/1665

**Comment by lahirusamith** [comment]
This PR is closing because same changes added in https://github.com/ballerina-platform/module-ballerinax-oracledb/pull/8 while retaining [gayal's PR](https://github.com/ballerina-platform/module-ballerinax-oracledb/pull/3) changes .
